R/tbl_custom_summary.R

```diff
@@ -284,7 +284,20 @@ tbl_custom_summary <- function(
   tbl_custom_summary_inputs <- as.list(environment())
 
   # checking function inputs ---------------------------------------------------
-  # verify if checks about stat_fns should be added at this stage
+  if (rlang::is_function(stat_fns)) {
+    stat_fns <-
+      inject(everything() ~ !!gts_mapper(stat_fns, "tbl_custom_summary(stat_fns=)"))
+  }
+  if (rlang::is_bare_formula(stat_fns))
+    stat_fns <- list(stat_fns)
+  if (!rlang::is_bare_list(stat_fns))
+    stop(paste0(
+      "'stat-fns' argument must be a formula or a list of formulas.\n",
+      "LHS of the formula is the variable specification, ",
+      "and the RHS is a function:\n",
+      "everything() ~ continuous_summary(\"var\")"
+    ), call. = FALSE)
```

Comment:
@larmarange 

The formula list will eventually be processed by `broom.helpers::.formula_list_to_named_list()`. Rather than adding these checks and messaging here, would it not be preferable to make the messaging better in that function instead?

One other thing to consider...I've made a syntax error with these lists many many times when I want to apply the same instructions to all variables. Should we add an input type to `broom.helpers::.formula_list_to_named_list(typeof = NULL)`. If a typeof argument is supplied we could do two things: 
1. If we had `broom.helpers::.formula_list_to_named_list(x, typeof = rlang::is_function)` and `rlang::is_function(x)` is TRUE, we auto convert this to `list(everything() ~ x)`.
2. Add a check that each element on the RHS of the formula satisfies `rlang::is_function(.)` 

What do you think?

Reply:
@ddsjoberg  Yes it could be a good idea to improve messaging in `broom.helpers::.formula_list_to_named_list()` with the possibility of indicating a certain type.

To adapt the error message, we should have a way to indicate what is expected on the RHS side, i.e. a number, a function, a text... Maybe having an option in `broom.helpers::.formula_list_to_named_list()` to pass a custom string that will be glued in the error message? or to adapt the examples indicated in the error message? In that case, it should be possible to make the error message more custom in other parts of `gtsummary`.

Regarding point 2, I do not know if it is feasible. Right now you can provide the name of a function, e.g. `"mean"` and it is working, but not sure where it is evaluated as a function. We just need to be sure that the element is already evaluated at that step.

Would you like to propose a PR in `broom.helpers`? You are more familiar than me with `.formula_list_to_named_list()` and with possible implications for `gtsummary`.

Thanks